src/phonehome.cpp

```diff
@@ -0,0 +1,98 @@
+#include "phonehome.h"
+#include <QDateTime>
+#include <QTime>
+
+using namespace adiscope;
+
+PHONEHOME_H
+
+PhoneHome::PhoneHome(QSettings *settings, Preferences *pref) :
+	ApiObject(), m_timestamp(0), m_versionsJson(""),
+	preferences(pref), settings(settings)
+{
+	setObjectName("phonehome");
+	load(*settings);
+}
+
+PhoneHome::~PhoneHome() {
+	save(*settings);
+}
+
+void PhoneHome::setPreferences(Preferences* preferences)
+{
+	PhoneHome::preferences = preferences;
+}
+
+void PhoneHome::versionsRequest(bool force)
+{
+	qint64 timeout = 24 * 60 * 60 * 1000; // 24 hours
+	//qint64 timeout = 5 * 1000;  // 5 seconds
+	qint64 now = QDateTime::currentMSecsSinceEpoch();
+
+	if(m_timestamp +  timeout < now  || force)
+	{
+		// from swdownloads
+		QNetworkAccessManager* manager = new QNetworkAccessManager();
+		connect(manager, &QNetworkAccessManager::finished, this, &PhoneHome::onVersionsRequestFinished);
+		connect(manager, &QNetworkAccessManager::finished, manager, &QNetworkAccessManager::deleteLater);
+
+		manager->get(QNetworkRequest(QUrl(preferences->getCheck_updates_url())));
+	}
+	else
+	{
+		// from cache
+		if(!m_versionsJson.isEmpty())
+		{
+			QJsonDocument doc = QJsonDocument::fromJson(m_versionsJson.toUtf8());
+			extractVersionStringsFromJson(doc);
+		}
+		Q_EMIT(checkUpdatesFinished(m_timestamp));
+	}
+}
+
+void PhoneHome::extractVersionStringsFromJson(QJsonDocument doc)
```

Comment:
Do we need to copy the `QJsonDocument`, if not we can pass it like `const QJsonDocument &`